torch/csrc/jit/tensorexpr/analysis.h

```diff
@@ -46,6 +47,102 @@ class NodeFinder : public IRVisitor {
   std::vector<Node*> nodes;
 };
 
+// A class that analyzes the given program relevant for Block backend.
+class BlockAnalysis : public IRVisitor {
+ public:
+  bool is_buf_store_target(const Buf* buf) const {
+    return store_targets_.count(buf) > 0;
+  }
+
+  std::unordered_set<const Buf*> reads() const {
+    return loads_;
+  }
+
+  std::unordered_set<const Buf*> stores() const {
+    return store_targets_;
+  }
+
+  int block_size() const {
+    return block_size_;
+  }
+
+  void analysis_ran() {
+    analysis_runs_++;
+    return;
+  }
+
+  bool areBufsInMap(const std::unordered_set<const Buf*>& bufs) const {
+    for (auto const& arg : bufs) {
+      auto got = map_input_to_tensor_bufs_.find(arg->name_hint());
+      if (got == map_input_to_tensor_bufs_.end()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  const Buf* getMultiDimBuf(const Buf* buf) const {
+    auto input_ = map_input_to_tensor_bufs_.find(buf->name_hint());
+    if (input_ != map_input_to_tensor_bufs_.end()) {
+      return input_->second;
+    } else {
+      throw std::runtime_error("BlockCodeGen: Entry not in input/Buffer map");
+    }
+  }
+
+  std::string getInputName(const Buf* buf) const {
+    auto input_ = map_input_to_tensor_bufs_.find(buf->name_hint());
+    return input_->second->name_hint();
+  }
+
+  std::string getFlatInputName(const Buf* buf) const {
+    return getInputName(buf) + "_flat";
+  }
+
+
+ private:
+  void visit(const Store* v) override {
+    if (analysis_runs_) {
```

Comment:
It seems that you are only using analysis_runs_ as a boolean flag. In that case, it is better to use a "bool analysis_has_run_" flag, instead of a counter. A counter could overflow. Although the likelihood is low, it is better to not have it by design.

Reply:
Thanks! I did not need this variable anymore (to keep track of anlysis passes).  I removed it.